### ISSUE TITLE
[$500] [Due for payment 2025-02-13] [Add onboarding for new workspaces being created] Create new setup tasks when users create a workspace

### ISSUE THREAD

**Issue by francoisl** [Expensify/App#53509]
## Part of the project "Adding onboarding for new workspaces being created"
Main issue: https://github.com/Expensify/App/issues/48591
Doc section: https://docs.google.com/document/d/19u6ZQbNuF-v5oovIkQlmwYV4SXeAgey50pymsj5QHRY/edit?tab=t.0#heading=h.vwt3yefp8jmh (External only)

### Feature Description
When a user who didn't choose the option "Manage my team's expenses" during the initial onboarding eventually creates a workspace, create onboarding tasks to teach them what to do next.

### Implementation steps
1. Prep work:
   1. Add new constant `newDotTrackWorkspace` to [`backendOnboardingChoices`](https://github.com/Expensify/App/blob/acf418ced84cc7066476d8b7cce7fd52b8397f01/src/CONST.ts#L72) - [Design doc section](https://docs.google.com/document/d/19u6ZQbNuF-v5oovIkQlmwYV4SXeAgey50pymsj5QHRY/edit?tab=t.0)
   2. Extract the logic to generate the request and optimistic data for onboarding tasks out of `completeOnboarding()`, for clarity and reusability – [Design doc section](https://docs.google.com/document/d/19u6ZQbNuF-v5oovIkQlmwYV4SXeAgey50pymsj5QHRY/edit?tab=t.0#heading=h.ubtcarhcxvd4)

2. Create new setup tasks when users create a workspace
    - [Design doc section ('External' section only)](https://docs.google.com/document/d/19u6ZQbNuF-v5oovIkQlmwYV4SXeAgey50pymsj5QHRY/edit?tab=t.0#heading=h.z586f9pqzs6g)
    - Update `CategorizeTrackedExpense`, `ShareTrackedExpense`, and `CreateWorkspace` to generate and pass the new onboarding task data to the API if the user doesn't already have a `createWorkspace` onboarding task

<details><summary>Issue Owner</summary>Current Issue Owner: @francoisl</details>

**Comment by francoisl** [comment]
(Keeping internal because we're going to work with a C+ on this)

**Comment by mananjadhav** [comment]
Reviewed the design document. Will take another pass to see if there are any gaps. Will start with the implementation by tomorrow.

**Comment by mananjadhav** [comment]
@francoisl Is there anything pushed on the backend to start this?

**Comment by francoisl** [comment]
One part is deployed, another one won't be until at least tomorrow, and the tasks won't be saved until both parts are.

For now, the part 1 "Prep work" I described above shouldn't require any backend changes. Also, for step 2 we can start saving the task data in the optimistic data of the API request, and test it offline?

**Comment by francoisl** [comment]
@mananjadhav the backend changes for passing `guidedSetupData` and `engagementChoice` to the API commands `CategorizeTrackedExpense`, `ShareTrackedExpense`, and `CreateWorkspace` are now live. Let me know if anything is not working as expected.

**Comment by mananjadhav** [comment]
Okay thanks. I'll check those and reach out to you if any questions.

**Comment by mananjadhav** [comment]
Started with this, the PR will be up by tomorrow.

**Comment by mananjadhav** [comment]
I was out sick yesterday. I am half done, will finish this by weekend.

**Comment by anmurali** [comment]
@mananjadhav - do you have an update on this PR?

**Comment by mananjadhav** [comment]
Apologies I was unwell. I am just getting back and will have this ready in 1-2 days.

**Comment by danielrvidal** [comment]
Thank you @mananjadhav, let us know if you need anything from our side!

**Comment by mananjadhav** [comment]
PR work in progress.  Will be raising the draft today.

**Comment by francoisl** [comment]
Sounds good, feel free to send it my way even if it's not fully ready, I can start taking a look and testing.

**Comment by anmurali** [comment]
@ishpaul777 have you gotten started with a PR?

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/pull/55302 unfortunately needs to be redone. More details in slack: https://expensify.slack.com/archives/C01GTK53T8Q/p1738286793485059?thread_ts=1738272457.156859&cid=C01GTK53T8Q

**Comment by francoisl** [comment]
@mallenexpensify FYI I increased the payout since this was a little bigger than a regular bug fix / task.

**Comment by mallenexpensify** [comment]
@ishpaul777 can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~021890561801107228170

I see @mananjadhav assigned but I don't see a full review [on the PR ](https://www.upwork.com/jobs/~021890561801107228170)so I'm assuming payment isn't due to them. Comment if ya disagree.

**Comment by mananjadhav** [comment]
@mallenexpensify  The original [PR](https://github.com/Expensify/App/pull/55302#issuecomment-2625249358) was reviewed by me. But due to the https://github.com/Expensify/App/pull/55302#issuecomment-2626099622 we had to redo the PR. The new duplicate PR was merged where you won't see my review. Hence I am eligible for the payout.

**Comment by francoisl** [comment]
Yes that's correct ^. We had to revert the original PR due to some unrelated issues with bad commits being on the `main` branch, and then make the second identical PR. I directly approved and merged that second PR since it was the same as the first PR, which had already been reviewed and tested.

**Comment by mallenexpensify** [comment]
Contributor: @ishpaul777 paid $500 via Upwork
Contributor+: @mananjadhav due $500 via NewDot

Where are we at with the need for a regression test here? We don't have `Bug` or `New Feature` so a checklist wasn't auto-populated.

**Comment by garrettmknight** [comment]
$500 approved for @mananjadhav

**Comment by mananjadhav** [comment]
Considering this is a new feature does the checklist apply? and with regards to the regression test, would it already exist as a part of the feature design? If not I'll create one but I think in my previous projects experience, I've seen it exists for the new features.

**Comment by mallenexpensify** [comment]
We def want regression tests for new features, sometimes they're created all at once as part of a project and sometimes they're done per  feature.  @francoisl do you know how we want to handle here? Thx

**Comment by mananjadhav** [comment]
Quick bump @francoisl

**Comment by francoisl** [comment]
@mallenexpensify as far as I can tell, we just need to issue the payments and close this?

**Comment by mallenexpensify** [comment]
Going to wait a min for payment, pending potential regression. Note to self to check the regression timeline/window too.

**Comment by mananjadhav** [comment]
I cannot reproduce it either. 

> Going to wait a min for payment, pending potential regression. Note to self to check the regression timeline/window too.

@mallenexpensify This issue was supposed to be closed by Feb 13 and I think it's already paid out. I think we should close this issue and discuss in the linked issue.
